File: cpp/gabaritos/gabarito24.cpp
#include <iostream>
#include <vector>

class Fila {
private:
    std::vector<int> items;
public:
    void enqueue(int item) {
        items.push_back(item);
    }
    int dequeue() {
        if (!is_empty()) {
            int front = items.front();
            items.erase(items.begin());
            return front;
        }
        return -1; // erro
    }
    bool is_empty() {
        return items.empty();
    }
};

int main() {
    Fila fila;
    fila.enqueue(1);
    fila.enqueue(2);
    std::cout << "Dequeue: " << fila.dequeue() << std::endl;
    std::cout << "Vazia: " << (fila.is_empty() ? "Sim" : "Não") << std::endl;
    return 0;
}

File: csharp/gabaritos/gabarito24.cs
using System;
using System.Collections.Generic;

class Fila
{
    private List<int> items = new List<int>();

    public void Enqueue(int item)
    {
        items.Add(item);
    }

    public int? Dequeue()
    {
        if (!IsEmpty())
        {
            int item = items[0];
            items.RemoveAt(0);
            return item;
        }
        return null;
    }

    public bool IsEmpty()
    {
        return items.Count == 0;
    }
}

class Program
{
    static void Main()
    {
        Fila fila = new Fila();
        fila.Enqueue(1);
        fila.Enqueue(2);
        Console.WriteLine($"Dequeue: {fila.Dequeue()}");
        Console.WriteLine($"Vazia: {fila.IsEmpty()}");
    }
}